smack-core/src/main/java/org/jivesoftware/smack/ConnectionConfiguration.java

```diff
@@ -118,6 +118,8 @@
     private final String password;
     private final Resourcepart resource;
 
+    private final String language;
```

Comment:
This should be of type `java.util.Locale`.

Reply:
Isn't a locale more than just the language? Are we going to have issues of ambiguity here?

Reply:
> Isn't a locale more than just the language?

"A locale" yes, it describes how your date is formatted or your preferred input method

But think of `java.util.Locale` as a wrapper around BCP 47, which is what `xml:lang` also uses.

> Are we going to have issues of ambiguity here?

I don't see any.